google/cloud/firestore_v1/async_collection.py

```diff
@@ -119,7 +117,8 @@ async def list_documents(self, page_size=None):
             },
             metadata=self._client._rpc_metadata,
         )
-        return (_item_to_document_ref(self, i) for i in iterator)
+        async for i in iterator:
+            yield _item_to_document_ref(self, i)
```

Comment:
Prior to this change `list_documents` was used like `[i async for i in await list_documents]`. This change simplifies the use case to `[i async for i in list_documents]`. Previous type was a coroutine that returned an async generator, current type is just an async generator.

Alternatively, we could compose the entire list and return just that, though that seems counterproductive.

Reply:
I would suggest we use a generator. ( `( )` instead of `[ ]` )

Reply:
Resolved in chat. Using a generator results in a different return type, which complicates the use case. In the case that we do not explicitly use yield, we have a coroutine that returns an async generator, which must be awaited on before using `async for`. If we explicitly iterate over the results and yield, the type is an async generator which we can directly use `async for` on.